pkg/openapi/server.spec.yaml

```diff
@@ -189,6 +278,59 @@ components:
           type: string
         workloadPools:
           $ref: '#/components/schemas/computeClusterWorkloadPools'
+    computeClusterStatus:
+      description: Compute cluster status.
+      type: object
+      properties:
+        workloadPools:
+          $ref: '#/components/schemas/computeClusterWorkloadPoolsStatus'
+    computeClusterWorkloadPoolsStatus:
+      description: A list of Compute cluster workload pools status.
+      type: array
+      items:
+        $ref: '#/components/schemas/computeClusterWorkloadPoolStatus'
+    computeClusterWorkloadPoolStatus:
+      description: Compute cluster workload pool status.
+      type: object
+      required:
+      - name
+      properties:
+        name:
+          description: Workload pool name.
+          type: string
+        machines:
+          $ref: '#/components/schemas/computeClusterMachinesStatus'
+    computeClusterMachinesStatus:
+      description: A list of Compute cluster machines status.
+      type: array
+      items:
+        $ref: '#/components/schemas/computeClusterMachineStatus'
+    computeClusterMachineStatus:
+      description: Compute cluster machine status.
+      type: object
+      required:
+      - hostname
+      - privateIP
+      - status
+      properties:
+        hostname:
+          description: Machine hostname.
+          type: string
+        privateIP:
+          description: Machine private IP address.
+          type: string
+        publicIP:
+          description: Machine public IP address.
+          type: string
+        status:
```

Comment:
You should reference https://github.com/unikorn-cloud/core/blob/main/pkg/openapi/common.spec.yaml#L90 here to keep things in sync.